stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/GiphyMediaAttachmentView.kt

```diff
@@ -0,0 +1,217 @@
+package io.getstream.chat.android.ui.message.list.adapter.view.internal
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.annotation.Px
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import androidx.core.view.updateLayoutParams
+import com.getstream.sdk.chat.images.load
+import com.getstream.sdk.chat.images.loadAndResize
+import com.getstream.sdk.chat.model.ModelType
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import com.google.android.material.shape.ShapeAppearanceModel
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
+import io.getstream.chat.android.ui.common.extensions.internal.createStreamThemeWrapper
+import io.getstream.chat.android.ui.common.extensions.internal.dpToPx
+import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
+import io.getstream.chat.android.ui.databinding.StreamUiGiphyMediaAttachmentViewBinding
+import io.getstream.chat.android.ui.message.list.adapter.view.GiphyMediaAttachmentViewStyle
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * View used to display Giphy images inside a shapeable container.
+ */
+public class GiphyMediaAttachmentView : ConstraintLayout {
```

Comment:
Do we want to publicly expose this view?

Reply:
In case our users want to use this in their custom UI, I would say yes.

Doesn't allow much control to the outside, could limit this even further